### PR TITLE
Added Get Cluster list and Primary Cluster function with unit test co…

### PR DESCRIPTION
**Pull request by jain-prerna**
…verage

### CODE REVIEW

**Review by joechainz — DISMISSED** [review]

File: src/saltext/vmware/modules/vmc_sddc_clusters.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+log = logging.getLogger(__name__)
+
+__virtualname__ = "vmc_sddc_clusters"
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def get(hostname, refresh_key, authorization_host, org_id, sddc_id, verify_ssl=True, cert=None):
```
Comment (waynew) [inline src/saltext/vmware/modules/vmc_sddc_clusters.py:21]: **suggestion** should be `list_` instead

**Review by waynew — CHANGES_REQUESTED** [review]
Are there integration tests for these two modules?

File: src/saltext/vmware/modules/vmc_sddc_clusters.py (hunk 1 above)
Comment (jain-prerna) [reply src/saltext/vmware/modules/vmc_sddc_clusters.py:21]: https://github.com/saltstack/salt-ext-modules-vmware/pull/221#discussion_r845685289

**Comment by jain-prerna** [comment]
> Are there integration tests for these two modules?

yes , in next PR i will raise other funcitons and test coverage

**Comment by waynew** [comment]
If the aim is to reduce the amount of code in a review I appreciate the motivation. However, that's a misguided way to split things up.

Tests should always accompany code. It can be good to separate functionality, for example - with the CRUD style PRs that we have they could be created like this:

1. First PR, create, and either get or list - some way to verify that the create worked.
2. Second PR, list or get - whichever one wasn't part of the first PR
3. Third PR, update (if possible)
4. Final PR, delete

Each PR should include comprehensive unit & functional tests for the operations in the PR.

**Review by waynew — CHANGES_REQUESTED** [review]
Before I can finish reviewing this:

- [ ] - Add integration test for ~get~ list
- [ ] - Add integration test for `get_primary`~`_cluster`~

It's OK to create a new PR with the create+tests and then rebase this PR on top of that along with the necessary integration tests for this PR - I can reveiw+merge the create functionality first.

File: src/saltext/vmware/modules/vmc_sddc_clusters.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        cert=cert,
+    )
+    if "error" in sddc_detail:
+        return sddc_detail
+    cluster_details = sddc_detail["resource_config"]["clusters"]
+    result = {"description": "vmc_sddc_clusters.get", "clusters": cluster_details}
+    return result
+
+
+def get_primary_cluster(
```
[the remark this answers is not in the document]
Comment (jain-prerna) [reply src/saltext/vmware/modules/vmc_sddc_clusters.py:75]: sure, i made this change

**Comment by jain-prerna** [comment]
I have raised another PR with IT coverage https://github.com/saltstack/salt-ext-modules-vmware/pull/251
But i need to discuss a scenario with you there before merging the PR

Creation of cluster takes long enough time,  so when it reaches to deletion step , the error is thrown
`cluster is currently not in a state where it can be deleted. Please try once the status is READY or FAILED`

**Pull request closed without merging**